src/helm/benchmark/scenarios/melt_scenarios.py

```diff
@@ -0,0 +1,349 @@
+        self.dataset_name = dataset_name
+        self.revision = revision
+        self.subset = subset
+        self.train_min_length = train_min_length
+        self.train_max_length = train_max_length
+        self.doc_max_length = doc_max_length
+        self.article_key = article_key
+        self.summary_key = summary_key
+        self.splits = splits
+
+    def _clean_and_truncate(self, text: str, max_length: Optional[int] = None) -> str:
+        return " ".join(text.split()[:max_length])
+
+    def get_instances_for_splits(self, splits: Dict[str, str]) -> List[Instance]:
+        """
+        Helper for generating instances for a split.
+        Args:
+            splits (dict): Which splits to partition the data into.
+        Returns:
+            List[Instance]: Instances from the file for the specified split.
+        """
+        instances: List[Instance] = []
+        dataset = load_dataset(
+            self.dataset_name,
+            self.subset,
+            revision=self.revision,
+            trust_remote_code=True,
+        )
+
+        for dataset_split_name, helm_split_name in splits.items():
+            if dataset_split_name not in dataset:
+                raise ValueError(f"Could not find split {dataset_split_name} in dataset {self.dataset_name}")
+
+            for sample in dataset[dataset_split_name]:
+                article: str = self._clean_and_truncate(sample[self.article_key], self.doc_max_length)
+                summary: str = self._clean_and_truncate(sample[self.summary_key])
+
+                if helm_split_name == "train":
+                    art_len = len(article.split())
+                    if self.train_max_length and art_len > self.train_max_length:
+                        continue
+                    if self.train_min_length and art_len < self.train_min_length:
+                        continue
+
+                instances.append(
+                    Instance(
+                        input=Input(text=article),
+                        references=[Reference(Output(text=summary), tags=[CORRECT_TAG])],
+                        split=helm_split_name,
+                    )
+                )
+
+        return instances
+
+    def get_instances(self, output_path: str) -> List[Instance]:
+        if self.splits is None:
+            splits = {"train": TRAIN_SPLIT, "validation": VALID_SPLIT, "test": TEST_SPLIT}
+        else:
+            splits = {}
+            if "train" in self.splits:
+                splits[self.splits[TRAIN_SPLIT]] = TRAIN_SPLIT
+            if "validation" in self.splits:
+                splits[self.splits[VALID_SPLIT]] = VALID_SPLIT
+            if "test" in self.splits:
+                splits[self.splits[TEST_SPLIT]] = TEST_SPLIT
+
+        instances: List[Instance] = self.get_instances_for_splits(splits=splits)
+        return instances
+
+
+class MELTSummarizationVietnewsScenario(MELTSummarizationScenario):
+    """
+    Scenario for summarization on Vietnews dataset.
```

Comment:
Add a more detailed description (e.g. which newspapers are the dataset from?) and a link to a paper if applicable.